aw-transform/src/flood.rs

```diff
@@ -65,14 +135,15 @@ pub fn flood(events: Vec<Event>, pulsetime: chrono::Duration) -> Vec<Event> {
                 // to the new_events vec
                 continue;
             } else {
-                if chrono::Duration::seconds(0) > gap {
+                if gap < chrono::Duration::seconds(0) {
                     if !warned_negative_gap_unsafe {
                         warn!("Gap was of negative duration ({}s) and could NOT be safely merged. This error will only show once per batch.", gap);
                         warned_negative_gap_unsafe = true;
                     }
+                } else {
+                    // Extend e1 to the middle between e1 and e2
+                    e1.duration = e1.duration + (gap / 2);
                 }
-                // Extend e1 to the middle between e1 and e2
-                e1.duration = e1.duration + (gap / 2);
```

Comment:
This was a major issue, as if the gap is negative that could lead to e1.duration getting shorter.

Reply:
I guess that's true, but if you have negative gaps the data is already unreliable as it is. So there's not much of an improvement, since the raw data is already "bad".

Reply:
It's intended to fix this: https://github.com/ActivityWatch/activitywatch/issues/602

Which I consider a significant improvement, as now it should handle it as well as aw-server-python.